### PR TITLE
Add example for Ray Serve deployment plugin

### PR DESCRIPTION
**Pull request by architkulkarni**
## What changes are proposed in this pull request?

This PR adds a basic example of using the MLflow Ray Serve deployment plugin.  In the example, a model is trained to classify the Iris dataset using `sklearn`. Next, the model is deployed on Ray Serve and scaled up.

This PR also adds the Ray Serve plugin to the list of known third-party deployment plugins in the MLflow documentation.

## How is this patch tested?

A test has been added alongside the rest of the example tests.
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
Add usage example for MLflow Ray Serve deployment plugin
### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by dmatrix** [comment]
@architkulkarni You will want to [sign your commits](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work), for the DCO test to succeed.

**Comment by architkulkarni** [comment]
Thanks @dmatrix, apologies for the oversight.  It's fixed now.

**Review by dmatrix — CHANGES_REQUESTED** [review]
Shouldn't that be

**Review by dmatrix — CHANGES_REQUESTED** [review]
Shouldn't this be the `-m model-uri?`

File: examples/ray_serve/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The plugin supports both a command line interface and a Python API.  Below we will use the command line interface.  For the full API documentation, see https://www.mlflow.org/docs/latest/cli.html#mlflow-deployments and https://www.mlflow.org/docs/latest/python_api/mlflow.deployments.html.
+
+## Plugin Installation
+
+Please follow the installation instructions for the Ray Serve deployment plugin: https://github.com/ray-project/mlflow-ray-serve
+
+## Instructions
+
+First, run `train_model.py`.  This trains and saves our classifier and sets up automatic logging to MLflow.  It also prints the mean squared error and the target names, which are species of iris:
```
[the remark this answers is not in the document]
Comment (architkulkarni) [reply examples/ray_serve/README.md:13]: Thanks, that makes sense!  It should be more readable now.

**Review by dmatrix — COMMENTED** [review]
LGTM with the sequence of steps.

File: examples/ray_serve/train_model.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    iris_dataset["target"],
+    iris_dataset["target_names"],
+)
+
+# Instantiate model
+model = GradientBoostingClassifier()
+
+# Split training and validation data
+np.random.shuffle(data)
+np.random.shuffle(target)
```
Comment (harupy) [inline examples/ray_serve/train_model.py:25]: Should we avoid shuffling `data` and `target` separately?

Comment (harupy) [reply examples/ray_serve/train_model.py:25]: I think we can use https://scikit-learn.org/stable/modules/generated/sklearn.utils.shuffle.html

File: examples/ray_serve/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+The plugin supports both a command line interface and a Python API.  Below we will use the command line interface.  For the full API documentation, see https://www.mlflow.org/docs/latest/cli.html#mlflow-deployments and https://www.mlflow.org/docs/latest/python_api/mlflow.deployments.html.
+
+## Plugin Installation
+
+Please follow the installation instructions for the Ray Serve deployment plugin: https://github.com/ray-project/mlflow-ray-serve
+
+## Instructions
+
+First, navigate to the directory for this example, `/mlflow/examples/ray_serve/`.
```
Comment (dmatrix) [inline examples/ray_serve/README.md:13]: should that be relative path `mlflow/examples/ray_serv/`

File: examples/ray_serve/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Next, set the MLflow Tracking URI environment variable to the location where the Model Registry resides:
+
+`export MLFLOW_TRACKING_URI=sqlite:///mlruns.db`
+
+Now start a Ray cluster with the following command:
+
+`ray start --head`
+
+Next, start a long-running Ray Serve instance on our Ray cluster:
```
Comment (dmatrix) [inline examples/ray_serve/README.md:29]: Should that pronoun me `your`? Our suggests it's maintained outside, elsewhere.

File: examples/ray_serve/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+`serve start`
+
+Ray Serve is now running and ready to deploy MLflow models.  The MLflow Ray Serve plugin features both a Python API as well as a command-line interface. For this example, we'll use the command line interface.
+
+Finally, we may deploy our model by creating an instance using the following command:
```
Comment (dmatrix) [inline examples/ray_serve/README.md:35]: Perhaps `can` is better than using tentative `may`

File: examples/ray_serve/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+`mlflow deployments create -t ray-serve -m models:/RayMLflowIntegration/1 --name iris:v1`
+
+The `-t` parameter here is the deployment target, which in our case is Ray Serve.  The `-m` parameter is the Model URI, which consists of the model name and model version.
```
Comment (dmatrix) [inline examples/ray_serve/README.md:39]: maybe say:  which consists of the registered model name and version in the Model Registry.

File: examples/ray_serve/train_model.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+from sklearn.datasets import load_iris
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.metrics import mean_squared_error
+from sklearn.utils import shuffle
+
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+
+
+def print_model_info(m_uri, x):
```
Comment (dmatrix) [inline examples/ray_serve/train_model.py:15]: I know I wrote this code bit. Perhaps call this function `eval_and_print_model_info(...)`

**Review by dmatrix — COMMENTED** [review]
Added minor nits and suggestions.

**Comment by architkulkarni** [comment]
Thanks a lot for the detailed review!  I implemented the suggestions and got rid of `print_model_info` since it isn't used in the example and I forgot to remove it earlier.

**Review by dmatrix — APPROVED** [review]
LGTM :-) after the test failures have been resolved.

@harupy If all looks good to you, we can merge it. Not sure if it's too late for the 1.14 release. But will leave that up to you.

**Review by harupy — APPROVED** [review]
LGTM :)

**Comment by harupy** [comment]
@architkulkarni The flavors check should be fixed by merging the master branch.

**Comment by architkulkarni** [comment]
@harupy Thanks, it worked!
